### ISSUE TITLE
about ACC on ImageNet-LT

### ISSUE THREAD

**Issue by ruikangliu** [Keke921/GCLLoss#2]
Thanks for your work and sharing your codes! I reproduced your code using the repo provided by the paper "Decoupling Representation and Classifier for Long-Tailed Recognition" and trained GCLLoss on ImageNet-LT (90 epochs for 1st stage and 10 epochs for 2nd stage; 1080Ti x 4). However, I only got 49.3% accuracy without mixup and 42.6% accuracy with mixup, which is much lower than 54.88% accuracy in the paper.

I leave all the hyperparameters in GCLLoss and CBEN sampler the same as the code you provided:
1st stage:
{
      "m": 0.0,
      "s": 30,
      "noise_mul": 0.5,
      "gamma": 0.0,

      "a": 0.9999
      "b": 1e-5
}
2nd stage:
{
      "m": 0.1,
      "s": 20,
      "noise_mul": 0.5,
      "gamma": 1.0

      "a": 0.9999
      "b": 1e-5
}

Could you analyze the problem about this or provide more details about experimental settings on ImageNet_LT and iNat18 such as epochs for 1st stage and 2nd stage? Thanks a lot.

**Comment by Keke921** [comment]
> Thanks for your work and sharing your codes! I reproduced your code using the repo provided by the paper "Decoupling Representation and Classifier for Long-Tailed Recognition" and trained GCLLoss on ImageNet-LT (90 epochs for 1st stage and 10 epochs for 2nd stage; 1080Ti x 4). However, I only got 49.3% accuracy without mixup and 42.6% accuracy with mixup, which is much lower than 54.88% accuracy in the paper.
> 
> I leave all the hyperparameters in GCLLoss and CBEN sampler the same as the code you provided: 1st stage: { "m": 0.0, "s": 30, "noise_mul": 0.5, "gamma": 0.0,
> 
> ```
>   "a": 0.9999
>   "b": 1e-5
> ```
> 
> } 2nd stage: { "m": 0.1, "s": 20, "noise_mul": 0.5, "gamma": 1.0
> 
> ```
>   "a": 0.9999
>   "b": 1e-5
> ```
> 
> }
> 
> Could you analyze the problem about this or provide more details about experimental settings on ImageNet_LT and iNat18 such as epochs for 1st stage and 2nd stage? Thanks a lot.

Sorry for the trouble. I reproduced all the methods (the comparison methods and GCL) listed in the paper with 200 eppochs.

The result  '42.6% accuracy with mixup' is strange. I'm not sure if this is normal. My intuition is that mixup is good for feature extraction. Therefore, it should be better to use mixup in the stage1. The stage 2 might be better without mixup. I will test this experimentally.

The details for imagenet are below:
1st stage:
{
      "batch_size": 256,
      "lr":0.1,
      “epoch”: 200,
      "m": 0.0,
      "s": 30,
      "noise_mul": 1,
      "gamma": 0.0,   
      "a": 0.9999
      "b": 1e-5
}
2nd stage:
{
      "batch_size": 256,
      "lr":0.1,
      “epoch”: 10,
      "m": 0.1,
      "s": 20,
      "noise_mul": 1,
      "gamma": 1.0   #may be useless, I do not use the focal loss actually. 
      "a": 0.9999
      "b": 1e-5
}
Both stages use mixup. 

The Acc. of stage1 is: 51.732
The Acc. of stage2 is: 54.884

The details for inat:
1st stage:
{
      "batch_size": 256,
      "lr":0.1,
      “epoch”: 200,
      "m": 0.0,
      "s": 30,
      "noise_mul": 0.8,
      #"gamma": 0.0,    
      "a": 0.9999
      "b": 1e-5
}
2nd stage:
{
      "batch_size": 256,
      "lr":0.1,
      “epoch”: 30,
      "m": 0.1,
      "s": 25,
      "noise_mul": 0.8,
      "gamma": 1.0   #may be useless, I do not use the focal loss actually. 
      "a": 0.9999
      "b": 1e-5
}

Acc:
stage1: 70.327
stage2: 72.005

Hope this can be helpful. Thanks for attention.

**Comment by ruikangliu** [comment]
Thanks for your reply! I will have a try with the new settings.

**Comment by ruikangliu** [comment]
> > Thanks for your work and sharing your codes! I reproduced your code using the repo provided by the paper "Decoupling Representation and Classifier for Long-Tailed Recognition" and trained GCLLoss on ImageNet-LT (90 epochs for 1st stage and 10 epochs for 2nd stage; 1080Ti x 4). However, I only got 49.3% accuracy without mixup and 42.6% accuracy with mixup, which is much lower than 54.88% accuracy in the paper.
> > I leave all the hyperparameters in GCLLoss and CBEN sampler the same as the code you provided: 1st stage: { "m": 0.0, "s": 30, "noise_mul": 0.5, "gamma": 0.0,
> > ```
> >   "a": 0.9999
> >   "b": 1e-5
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > } 2nd stage: { "m": 0.1, "s": 20, "noise_mul": 0.5, "gamma": 1.0
> > ```
> >   "a": 0.9999
> >   "b": 1e-5
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > }
> > Could you analyze the problem about this or provide more details about experimental settings on ImageNet_LT and iNat18 such as epochs for 1st stage and 2nd stage? Thanks a lot.
> 
> Sorry for the trouble. I reproduced all the methods (the comparison methods and GCL) listed in the paper with 200 eppochs.
> 
> The result '42.6% accuracy with mixup' is strange. I'm not sure if this is normal. My intuition is that mixup is good for feature extraction. Therefore, it should be better to use mixup in the stage1. The stage 2 might be better without mixup. I will test this experimentally.
> 
> The details for imagenet are below: 1st stage: { "batch_size": 256, "lr":0.1, “epoch”: 200, "m": 0.0, "s": 30, "noise_mul": 1, "gamma": 0.0, "a": 0.9999 "b": 1e-5 } 2nd stage: { "batch_size": 256, "lr":0.1, “epoch”: 10, "m": 0.1, "s": 20, "noise_mul": 1, "gamma": 1.0 #may be useless, I do not use the focal loss actually. "a": 0.9999 "b": 1e-5 } Both stages use mixup.
> 
> The Acc. of stage1 is: 51.732 The Acc. of stage2 is: 54.884
> 
> The details for inat: 1st stage: { "batch_size": 256, "lr":0.1, “epoch”: 200, "m": 0.0, "s": 30, "noise_mul": 0.8, #"gamma": 0.0, "a": 0.9999 "b": 1e-5 } 2nd stage: { "batch_size": 256, "lr":0.1, “epoch”: 30, "m": 0.1, "s": 25, "noise_mul": 0.8, "gamma": 1.0 #may be useless, I do not use the focal loss actually. "a": 0.9999 "b": 1e-5 }
> 
> Acc: stage1: 70.327 stage2: 72.005
> 
> Hope this can be helpful. Thanks for attention.

And by the way, could you share your .pth file and training log on ImageNet-LT and iNat18?

**Comment by Keke921** [comment]
> > > Thanks for your work and sharing your codes! I reproduced your code using the repo provided by the paper "Decoupling Representation and Classifier for Long-Tailed Recognition" and trained GCLLoss on ImageNet-LT (90 epochs for 1st stage and 10 epochs for 2nd stage; 1080Ti x 4). However, I only got 49.3% accuracy without mixup and 42.6% accuracy with mixup, which is much lower than 54.88% accuracy in the paper.
> > > I leave all the hyperparameters in GCLLoss and CBEN sampler the same as the code you provided: 1st stage: { "m": 0.0, "s": 30, "noise_mul": 0.5, "gamma": 0.0,
> > > ```
> > >   "a": 0.9999
> > >   "b": 1e-5
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > } 2nd stage: { "m": 0.1, "s": 20, "noise_mul": 0.5, "gamma": 1.0
> > > ```
> > >   "a": 0.9999
> > >   "b": 1e-5
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > }
> > > Could you analyze the problem about this or provide more details about experimental settings on ImageNet_LT and iNat18 such as epochs for 1st stage and 2nd stage? Thanks a lot.
> > 
> > 
> > Sorry for the trouble. I reproduced all the methods (the comparison methods and GCL) listed in the paper with 200 eppochs.
> > The result '42.6% accuracy with mixup' is strange. I'm not sure if this is normal. My intuition is that mixup is good for feature extraction. Therefore, it should be better to use mixup in the stage1. The stage 2 might be better without mixup. I will test this experimentally.
> > The details for imagenet are below: 1st stage: { "batch_size": 256, "lr":0.1, “epoch”: 200, "m": 0.0, "s": 30, "noise_mul": 1, "gamma": 0.0, "a": 0.9999 "b": 1e-5 } 2nd stage: { "batch_size": 256, "lr":0.1, “epoch”: 10, "m": 0.1, "s": 20, "noise_mul": 1, "gamma": 1.0 #may be useless, I do not use the focal loss actually. "a": 0.9999 "b": 1e-5 } Both stages use mixup.
> > The Acc. of stage1 is: 51.732 The Acc. of stage2 is: 54.884
> > The details for inat: 1st stage: { "batch_size": 256, "lr":0.1, “epoch”: 200, "m": 0.0, "s": 30, "noise_mul": 0.8, #"gamma": 0.0, "a": 0.9999 "b": 1e-5 } 2nd stage: { "batch_size": 256, "lr":0.1, “epoch”: 30, "m": 0.1, "s": 25, "noise_mul": 0.8, "gamma": 1.0 #may be useless, I do not use the focal loss actually. "a": 0.9999 "b": 1e-5 }
> > Acc: stage1: 70.327 stage2: 72.005
> > Hope this can be helpful. Thanks for attention.
> 
> And by the way, could you share your .pth file and training log on ImageNet-LT and iNat18?

It's fine. But I have modified the code several times. The code in this repository may need to be modified to match the .pth file.  Since I left my previous workplace, there is no GPU retraining model now. I temporarily share previously stored .pth file for the model. Sorry for the inconvenience.
